File: dotnet/HotelBooking/HotelBooking/Models/Customer.cs
namespace HotelBooking.Models;

public class Customer
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    
    public string Email { get; set; }
    
    public List<Booking> Bookings { get; set; } = new();
}

public class CustomerWithTotalPayment
{
    public Customer Customer { get; set; }
    public int Amount { get; set; }
}

File: java/hotelbooking/src/main/java/no/egde/hotelbooking/models/Customer.java
package no.egde.hotelbooking.models;

import javax.persistence.*;
import java.util.List;

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private String email;

    @OneToMany
    @JoinColumn(name = "bookingId", referencedColumnName = "id")
    private List<Booking> bookings;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public List<Booking> getBookings() {
        return bookings;
    }

    public void setBookings(List<Booking> bookings) {
        this.bookings = bookings;
    }
}

